src/lib.rs

```diff
@@ -139,13 +139,35 @@ pub fn derive_pwrite(input: TokenStream) -> TokenStream {
     gen.into()
 }
 
-fn size_with(name: &syn::Ident) -> proc_macro2::TokenStream {
+fn size_with(name: &syn::Ident, fields: &syn::FieldsNamed) -> proc_macro2::TokenStream {
+    let items: Vec<_> = fields.named.iter().map(|f| {
+        let ty = &f.ty;
+        match ty {
+            &syn::Type::Array(ref array) => {
```

Comment:
Are there any other types besides array we need to do this with? Probably not, right?

Reply:
I know basically nothing about syn, I just copied this from another part of the file. :)

Reply:
I looked over [the syn docs](https://docs.rs/syn/0.15.0/syn/enum.Type.html) and while there are definitely other types that are legal to use there, I'm not sure that we gain anything by explicitly supporting them. Anything that implements `SizeWith<Endian>` should pretty much just work because the default case simply produces `<#ty>::size_with(ctx)`. In fact, once const generics stabilizes we could remove the special-case for arrays here and just add a generic impl like:
```rust
impl<Ctx, T: SizeWith<Ctx, Units=usize>, const N: usize> SizeWith<Ctx> for [T; N] {
  type Units = usize;
  #[inline]
    fn size_with(ctx: &Ctx) -> usize {
      <T>::size_with(ctx) * N
    }
}
```
...and then the default impl ought to work for arrays as well.